### PR TITLE
[AuthorityAggregator] EpochEnded to be retryable

### PR DESCRIPTION
**Pull request by akichidis**
## Description 

E2e localnet tests appear to fail sporadically due the the `EpochEnded` error. Although it should not be the common case it's possible to happen when a transaction has reached a node's RPC endpoint, but epoch changes in the middle. In this case an `EpochEnded` can be thrown when trying to access the AuthorityEpochStore. Making the error retryable will allow the quorum driver retry the transaction and hopefully succeed.

## Test plan 

CI

---

## Release notes

Check each box that your changes affect. If none of the boxes relate to your changes, release notes aren't required.

For each box you select, include information after the relevant heading that describes the impact of your changes that a user might notice and any actions they must take to implement updates. 

- [ ] Protocol: 
- [ ] Nodes (Validators and Full nodes): 
- [ ] Indexer: 
- [ ] JSON-RPC: 
- [ ] GraphQL: 
- [ ] CLI: 
- [ ] Rust SDK:
- [ ] REST API:

### CODE REVIEW

File: crates/sui-core/src/unit_tests/authority_aggregator_tests.rs
Comment (amnn) [inline crates/sui-core/src/unit_tests/authority_aggregator_tests.rs:1]: Could you also add a test case where one validator repeatedly responds with `EpochEnded` (i.e. is byzantine) and the other eventually recovers? That would confirm that "is_retryable" has the desired behaviour of only becoming relevant when we couldn't reach the quorum threshold.

**Review by amnn — COMMENTED** [review]
Just one additional comment for me, but I'm not the expert in these parts, so it would be good to get another reviewer's pair of :eyes: on this.

**Review by mwtian — APPROVED** [review]
This makes sense!

File: crates/sui-core/src/unit_tests/authority_aggregator_tests.rs
Comment (akichidis) [reply crates/sui-core/src/unit_tests/authority_aggregator_tests.rs:1]: The authority aggregator it self is not the one that is performing the retries, but the `quorum_driver` . I'll check if it's straightforward to do so, as this is now will be like testing an already existing feature of quorum driver (retry a tx with retriable errors) , and I'll do so.

**Pull request merged**